cmd/create/cluster/cmd.go

```diff
@@ -418,40 +399,89 @@ func run(cmd *cobra.Command, _ []string) {
 
 	// AWS ARN Role
 	roleARN := args.roleARN
-	if roleARN != "" && interactive.Enabled() {
+	if interactive.Enabled() {
+		help := cmd.Flags().Lookup("role-arn").Usage
+		if creator.IsSTS {
+			help = fmt.Sprintf("%s Since your AWS credentials are returning an STS ARN you can only "+
+				"create STS clusters. Otherwise, switch to IAM credentials.", help)
+		}
 		roleARN, err = interactive.GetString(interactive.Input{
 			Question: "Role ARN",
-			Help:     cmd.Flags().Lookup("role-arn").Usage,
+			Help:     help,
 			Default:  roleARN,
+			Required: creator.IsSTS,
```

Comment:
Here the Role ARN is marked as required if the customer uses STS credentials

Reply:
:+1: